lib/connections/limiter.go

```diff
@@ -124,6 +127,87 @@ func (lim *limiter) VerifyConfiguration(from, to config.Configuration) error {
 	return nil
 }
 
+func nextChange(to config.Configuration) time.Duration {
+	now := time.Now()
+
+	beginHour := to.Options.ScheduledRatesFromHour
+	beginMinute := to.Options.ScheduledRatesFromMinute
+	begin := time.Date(now.Year(), now.Month(), now.Day(), beginHour, beginMinute, 0, 0, now.Location())
+
+	tillBegin := begin.Sub(now)
+
+	var waitTime time.Duration
+
+	if tillBegin >= 0 {
+		waitTime = tillBegin
+	} else {
+		endHour := to.Options.ScheduledRatesToHour
+		endMinute := to.Options.ScheduledRatesToMinute
+		end := time.Date(now.Year(), now.Month(), now.Day(), endHour, endMinute, 0, 0, now.Location())
+		if end.Sub(begin) < 0 {
+			end = end.AddDate(0, 0, 1)
+		}
+		tillEnd := end.Sub(now)
+		if tillEnd >= 0 {
+			waitTime = tillEnd
+		} else {
+			begin = begin.AddDate(0, 0, 1)
+			waitTime = begin.Sub(now)
+		}
+	}
+
+	return waitTime
+}
+
+func (lim *limiter) timerLoop(timer *time.Timer, to config.Configuration) {
```

Comment:
I wouldn't pass the entire configuration, just the 4 values required.

Reply:
I didn't fix this one, in this function you are using 8 values (4 for time and 4 for rates), so I'm not sure it would be good change

Reply:
Just `config.Options` then?

Reply:
Sound good, changed